### ISSUE TITLE
bug: tailwind autocomplete not working

### ISSUE THREAD

**Issue by HakonHarnes** [LazyVim/LazyVim#1717]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.9.2

### Operating system/version

Arch Linux (kernel 6.5.6)

### Describe the bug

Tailwind autocomplete does not work when enabled through the `:LazyExtras`  command or in `lazy.lua`:

```
{ import = "lazyvim.plugins.extras.lang.tailwind" },
```

Opening an HTML file and running `:LspInfo` I see: 

```
 Language client log: /home/hakon/.local/state/nvim/lsp.log
 Detected filetype:   html
 
 0 client(s) attached to this buffer: 
 
 Other clients that match the filetype: html
 
 Config: tailwindcss
 	filetypes:         aspnetcorerazor, astro, astro-markdown, blade, clojure, django-html, htmldjango, edge, eelixir, elixir, ejs, erb, eruby, gohtml, gohtmltmpl, haml, handlebars, hbs, html, html-eex, heex, jade, leaf, liquid, mdx, mustache, njk, nunjucks, php, razor, slim, twig, css, less, postcss, sass, scss, stylus, sugarss, javascript, javascriptreact, reason, rescript, typescript, typescriptreact, vue, svelte
 	root directory:    Not found.
 	cmd:               /home/hakon/.local/share/nvim/mason/bin/tailwindcss-language-server --stdio
 	cmd is executable: true
 	autostart:         true
 	custom handlers:   
 
 Configured servers list: lua_ls, tailwindcss
```

Opening an TSX file in a React project and running `:LspInfo` I see: 

``` 
 Language client log: /home/hakon/.local/state/nvim/lsp.log
 Detected filetype:   typescriptreact
 
 1 client(s) attached to this buffer: 
 
 Client: tailwindcss (id: 1, bufnr: [3])
 	filetypes:       aspnetcorerazor, astro, astro-markdown, blade, clojure, django-html, htmldjango, edge, eelixir, elixir, ejs, erb, eruby, gohtml, gohtmltmpl, haml, handlebars, hbs, html, html-eex, heex, jade, leaf, liquid, mdx, mustache, njk, nunjucks, php, razor, slim, twig, css, less, postcss, sass, scss, stylus, sugarss, javascript, javascriptreact, reason, rescript, typescript, typescriptreact, vue, svelte
 	autostart:       true
 	root directory:  /home/hakon/print-buddy/frontend
 	cmd:             /home/hakon/.local/share/nvim/mason/bin/tailwindcss-language-server --stdio
 
 Configured servers list: tailwindcss, lua_ls
```

I have followed the installation steps in the docs and have deleted all cache/swap files before re-installing LazyVim.

### Steps To Reproduce

1. Install `extras.lang.tailwind`.
2. Open a HTML or TSX file and observe the autocomplete is not working. 

### Expected Behavior

Enabling `extras.lang.tailwind` should work.

**Comment by dpetka2001** [comment]
This has probably something to do with your LSP configuration. In a Lazyvim starter template, while enabling only the extra for tailwind from `LazyExtras`, when opening a `.html` file the LSP tailwind attaches correctly and auto-completion works as expected.

**Comment by folke** [comment]
Yep, same here. Works in html and tsx files. We also don't do any special config, so must be something on your end.

**Comment by HakonHarnes** [comment]
Okay. How would I go about debugging something like this? I have already deleted everything related to neovim, and my config is completely untouched. Running the following commands: 

```
rm -rf ~/.config/nvim ~/.local/share/nvim ~/.local/state/nvim ~/.cache/nvim 
git clone https://github.com/LazyVim/starter ~/.config/nvim
```

Then updating LazyVim, enabling only tailwind using the `:LazyExtras` command, and restarting neovim the issue persists.

**Comment by dpetka2001** [comment]
Tailwind (and every LSP server) searches for patterns to determine your `root_dir`. You can see the order of patterns it searches for in the corresponding `tailwind lspconfig docs](https://github.com/neovim/nvim-lspconfig/blob/8f3ddc448769f563248654a5099c943c7139137e/lua/lspconfig/server_configurations/tailwindcss.lua#L102-L110). I believe if you take a similar approach to issue #1307, it might help you (but I don't anything about your file structure so can't be sure). Maybe setting tailwind searching first for `.git` and then for the rest would be a solution to your problem or some other pattern that is distinct at the root of your project.

**Comment by m-wynn** [comment]
I've had a few issues with this language server not working.  I had it today and it was accompanied by an error like:

```
Error executing vim.schedule lua callback: /path/to/nvim/runtime/lua/vim/lsp.lua:1314: RPC[Error] code_name = InternalError, message = "Request initialize failed with message: Bad file descriptor"
```

In my case, this was due to files _anywhere_ in the project that are not owned by my user.  For me, I was launching docker and mounting a directory into a container running as a different UID.  Once I fixed that, I had completion again.
